python/ray/util/iter.py

```diff
@@ -193,10 +193,36 @@ def _with_transform(self, local_it_fn, name):
             name=self.name + name,
             parent_iterators=self.parent_iterators)
 
+    def transform(self, fn: Callable[[Iterable[T]], Iterable[U]]
+                  ) -> "ParallelIterator[U]":
+        """Remotely transform the iterator.
+
+        This is advanced version of for_each that allows you to apply arbitrary
+        generator transformations over the iterator. Prefer to use .for_each()
+        when possible for simplicity.
+
+        Args:
+            fn (func): function to use to transform the iterator. The function
```

Comment:
Can we specify that `fn` is called once per worker since this is stateful?

Reply:
Done.